File: leetcode/26/solution.py
"""
https://leetcode.com/problems/remove-duplicates-from-sorted-array/
26. Remove Duplicates from Sorted Array
"""
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, prev = 0, -101
        for c in nums:
            if prev < c:
                nums[i] = prev = c
                i += 1
        return i 

File: leetcode/26/solution.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut i = 0;
        let mut prev = -101;
        for j in 0..nums.len() {
            if prev < nums[j] {
                prev = nums[j];
                nums[i] = prev;
                i += 1;
            }
        }
        i as i32
    }
}
